controller/src/beerocks/master/tasks/optimal_path_task.cpp

```diff
@@ -198,16 +197,29 @@ void optimal_path_task::work()
             }
         }
 
-        // Check if hostapd has suitable ssid
+        // Check if hostap has suitable ssid
         auto it = potential_11k_aps.begin();
         while (it != potential_11k_aps.end()) {
-            if (database.get_hostap_vap_with_ssid(it->first, current_hostap_ssid).empty()) {
+
+            const std::string candidate_bssid =
+                database.get_hostap_vap_with_ssid(it->first, current_hostap_ssid);
+
+            if (candidate_bssid.empty()) {
                 LOG(INFO) << "Remove candidate " << it->first
                           << ". Hostap doesn't have current_hostap_ssid " << current_hostap_ssid;
                 it = potential_11k_aps.erase(it);
-            } else {
-                ++it;
+                continue;
+            }
+
+            // Steering allowed on all vaps unless load_steer_on_vaps list is defined
+            // on the platform , in that case verify that vap is on that list
+            if (!database.config.load_steer_on_vaps.empty() &&
+                !database.is_vap_on_steer_list(candidate_bssid)) {
+                TASK_LOG(INFO) << "Remove candidate " << it->first << " , vap " << candidate_bssid
+                               << " is not in steer list: " << database.config.load_steer_on_vaps;
+                it = potential_11k_aps.erase(it);
```

Comment:
Suggestion:
Maybe instead of removing it from the list, we will not add them in the first place?
This comment apply also to the cross candidates.

Reply:
Do you how many places I wanted to refractor the optimal task and held myself not to. You are probably right though... Next time 😎